AdyenDropIn/DropInComponent.swift

```diff
@@ -229,43 +225,95 @@ public final class DropInComponent: NSObject,
         return handler
     }()
 
-    internal func paymentMethodListComponent(onCancel: (() -> Void)?) -> PaymentMethodListComponent {
-        let paymentComponents = componentManager.sections
-        let component = PaymentMethodListComponent(
+    // ================= ROOT VIEW CONTROLLER ===============
+
+    internal lazy var rootViewController: UIViewController = {
+        if configuration.allowPreselectedPaymentView,
+           let preselectedComponent = componentManager.storedComponents.first {
+            let view = resolvePreselectedPaymentMethodView(for: preselectedComponent, onCancel: nil)
+            self.preselectedPaymentMethodView = view
+            return view
+        } else if configuration.allowsSkippingPaymentList,
+                  let singleRegularComponent = componentManager.singleRegularComponent {
+            setNecessaryDelegates(on: singleRegularComponent)
+            let componentView = resolveComponentView(from: singleRegularComponent)
+            self.componentView = componentView
+            return componentView
+        } else {
+            let view = resolvePaymentMethodListView(onCancel: nil)
+            self.paymentMethodListView = view
+            return view
+        }
+    }()
+
+    // ================= COMPONENT MODULE - ASSEMBLER ===============
+
+    internal func resolveComponentView(
+        from component: PresentableComponent
+    ) -> UIViewController {
+        let viewModel = ComponentViewModel(
+            component: component,
+            isRoot: false,
```

Comment:
A lot to unpack, I doubt if I should write such comment or not. I'd rather highlight this, but I guess it's just not feasible to do everything in one PR.

`isRoot` looks like a workaround, could be extracted to the current assembly layer that will make a decision and could maybe produce RootComponentViewModel instead?